### PR TITLE
Fix up midi stuff, support mild black midis

### LINKED ISSUE #1035: MIDI becomes garbled on other clients when the instrument moves outside the PVS
1. play an instrument on client A and walk away
2. instrument audio becomes extremely garbled on client B, timing is completely screwed when client A enters PVS range

We need to sanely destroy/create the renderer depending on whether the instrument is inside the client's PVS or not.

https://cdn.discordapp.com/attachments/310555209753690112/716599259272577024/midi-2020-05-31_12.28.57.webm

### PR DESCRIPTION
**Pull request by Tyler-IN**
Fixes https://github.com/space-wizards/space-station-14/issues/1035

Links to https://github.com/space-wizards/space-station-14/pull/1056

### CODE REVIEW

**Comment by Tyler-IN** [comment]
(Almost) unrelated changes;

Removed setter from TimeScale because it can't work this way.

BlockingCollection is an extra layer over ConcurrentQueue for the sync ctx, reduced to just ConcurrentQueue. Not significant.

Timestamp -> Tick to reduce implication that Timestamp is somehow seconds or milliseconds.

File: Robust.Client/Audio/Midi/MidiRenderer.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public interface IMidiRenderer : IDisposable
         /// <summary>
         ///     Gets or sets the Time Scale of the sequencer in ticks per second. Default is 1000 for 1 tick per millisecond.
         /// </summary>
-        double SequencerTimeScale { get; set; }
+        double SequencerTimeScale { get; }
```
Comment (gradientvera) [inline Robust.Client/Audio/Midi/MidiRenderer.cs]: Modify the doc to reflect this change.

File: Robust.Client/Audio/Midi/MidiManager.cs [hunk 2]
```diff
@@ -310,11 +310,22 @@ public void FrameUpdate(float frameTime)
...
+                        {
+                            // just move the source far away instead of to NaN
+                            if (renderer.Source.SetPosition(new Vector2(float.MaxValue, float.MaxValue)))
+                            {
+                                continue;
+                            }
+                        }
+
+                        _midiSawmill?.Warning("Interrupting positional audio, can't set position.");
+                        renderer.Source.StopPlaying();
```
Comment (gradientvera) [inline Robust.Client/Audio/Midi/MidiManager.cs]: Even if you handle NaN cases manually, the actual issue still persists: There's no way to effectively stop or pause a renderer's audio source playback. The renderer will just keep playing the audio source for as long as it renders audio.

**Review by gradientvera — CHANGES_REQUESTED** [review]

File: Robust.Client/Audio/Midi/MidiManager.cs (hunk 2 above)
Comment (Tyler-IN) [reply Robust.Client/Audio/Midi/MidiManager.cs]: Yeah, that's intentional here... fades in nicely from a distance and thru walls.

File: Robust.Client/Audio/Midi/MidiRenderer.cs (hunk 1 above)
Comment (Tyler-IN) [reply Robust.Client/Audio/Midi/MidiRenderer.cs]: Ok

File: Robust.Client/Audio/Midi/MidiRenderer.cs [hunk 3]
```diff
@@ -79,7 +79,7 @@ public interface IMidiRenderer : IDisposable
         /// <summary>
         ///     Gets or sets the Time Scale of the sequencer in ticks per second. Default is 1000 for 1 tick per millisecond.
```
Comment (Tyler-IN) [inline Robust.Client/Audio/Midi/MidiRenderer.cs]:
```suggestion
        ///     Gets the Time Scale of the sequencer in ticks per second. Default is 1000 for 1 tick per millisecond.
```
@Zumorica like this?

File: Robust.Client/Audio/Midi/MidiRenderer.cs (hunk 1 above)
Comment (Tyler-IN) [reply Robust.Client/Audio/Midi/MidiRenderer.cs]: See https://github.com/space-wizards/RobustToolbox/pull/1092#discussion_r432974248

File: Robust.Client/Audio/Midi/MidiRenderer.cs (hunk 3 above)
Comment (gradientvera) [reply Robust.Client/Audio/Midi/MidiRenderer.cs]: Yeah, exactly like that.

**Review by gradientvera — APPROVED** [review]
Renderers could still use a way to get paused, but other than that this seems okay.

**Pull request merged**